Rēķinot citas izteiksmes mainās:

*Pie (def) rindkopas mainās tikai - (a)

*Pie (while) rindkopas mainās tikai - (R - Rekurences reizinātājs)

Dotā funkcija: y = exp(-x*x)

No bibliotēkas (math) importē mums nepieciešamo (exp) funkciju:

Argumentam x ir piešķirta vērtība 0.56 un y definē kā exp(-x*x), izmantojot importēto (exp) funkciju un iegūstam funkcijas vērtību y(x):

Definēju savu funkciju un piešķiru vērtības argumentiem k (summas locekļu skaits),

a (attiecīgā soļa loceklis), S (summēšanā iegūtā vērtība), un, 

izprintēju 0-to funkcijas summēšanas locekli yy piešķiru mans_exp(x) vērtību.

Izveidoju ciklu, kurā tiek izmantoti 500 summas locekļi un attiecīgās f-jas aprēķinātais rekurences reizinātājs 
R = (-1*x**2/k), lai aprēķinātu f-jas vērtību. 

In [2]:
#_*_ coding: utf-8 -*-
from math import exp

def mans_exp(x):
    k = 0
    a = (-1)**k*x**(2*k)/(1)
    S = a
    print("a0 = %6.2f S0 = %6.2f"%(a,S))

    while k < 500:
        k = k + 1
        R = -1*x**2/k
        a = a * R
        S = S + a
        if  k == 499:
            print("a%d = %6.2f S%d = %6.2f"%(k,a,k,S))
        elif k == 500:
            print("a%d = %6.2f S%d = %6.2f"%(k,a,k,S))
    
    
    return S

print("exp(-x*x) aprekinasana: ")
x = float(input("Lietotāj, lūdzu ievadi skaitli argumentu (x): "))
y = exp(-x*x)
print("exp(-x*x) (%.2f) = %6.2f"%(x,y))

yy = mans_exp(x)
print("exp(-x*x) (%.2f) caur summu: %6.2f"%(x,yy))
                                                           

print(" ")
print("                500                               ")
print("               ______                             ")
print("               \               k      2*k         ")
print("                \          (-1)   * x             ")
print("     exp(-x*x) = >        _________________       ..., kur -\u221E < x < +\u221E  ")
print("                /                                 ")
print("               /_____           k!                ")
print("                 k=0                              ")
print("                                                  ")
print("                                                  ")
print("                                             2    ")
print("                                    (-1) * x      ")
print("     Rekurences reizinatajs:      ______________  ")
print("                                                  ")
print("                                         k        ")

exp(-x*x) aprekinasana: 
Lietotāj, lūdzu ievadi skaitli argumentu (x): 0.56
exp(-x*x) (0.56) =   0.73
a0 =   1.00 S0 =   1.00
a499 =  -0.00 S499 =   0.73
a500 =   0.00 S500 =   0.73
exp(-x*x) (0.56) caur summu:   0.73
 
                500                               
               ______                             
               \               k      2*k         
                \          (-1)   * x             
     exp(-x*x) = >        _________________       ..., kur -∞ < x < +∞  
                /                                 
               /_____           k!                
                 k=0                              
                                                  
                                                  
                                             2    
                                    (-1) * x      
     Rekurences reizinatajs:      ______________  
                                                  
                                         k  

Pēc summēšanas cikla izpildes 500-tais summēšanas loceklis ir 0.73, kas attiecīgi ir arī mans_exp(x) iegūtā vērtība, to arī izprintēju.

Pēc dotajiem norādījumiem izvadu arī summas izteiksmi un rekurences funkcionālā reizinātāja izteiksmi, izmantojot ASCII simbolus.